code/game/objects/structures/holosign.dm

```diff
@@ -121,11 +121,14 @@
 
 /obj/machinery/holosign/barrier/power_shield/Initialize(loc, source_projector)
 	. = ..()
+	var/area/a = get_area(src)
+	if(a.power_equip == FALSE)
+		stack_trace("Power shield created without power avaiable")
+		qdel(src)
```

Comment:
```suggestion
		stack_trace("Power shield created without power avaiable")
		qdel(src)
		return
```
You definitely don't want the continue with the proc after qdeleting.

Reply:
right, i'll change it